### ISSUE TITLE
BYO domain name

### ISSUE THREAD

**Issue by yo1dog** [mozilla/fx-private-relay#3379]
Love the idea of Firefox Relay, but I don't want to permanently tie myself to mozmail.com.

- What if Firefox Relay is discontinued?
- What if mozmail.com becomes blacklisted as it nearly did?

If I could use one of my own domain names (e.g. {random}@yo1dogrelay.com), I avoid both issues. If Relay is discontinued I could update the MX records to point to a different relay service, or directly to an email service such as gmail. And the unique domain name avoids blacklists.

-----

Aside: While theoretically these risks could also apply to  hotmail.com, outlook.com, icloud.com, etc., I feel mozmail.com is uniquely at risk because it is both relatively new and used only for a relay service. It does not have an extremely large set of "normal" inboxes behind it which makes the others nearly impervious to being shutdown or blacklisted.

**Comment by nilshoell** [comment]
I would love to see a feature like this, maybe in one of the new price tiers?
That would certainly increase the number of applicable use cases for Relay.

**Comment by rafalkrupinski** [comment]
Custom domains is a security anti-feature.

I really don't get why Moz is putting that huge banner bugging me to set one up.

**Comment by sosnik** [comment]
I disagree that it is a *security *anti-feature.
It may be a *privacy *one, in that the main point of alias domains is to
'hide within the crowd'. But it is not a security issue.
Not accommodating custom domains is also a vendor-lockin issue. Say I use
ff relay with 400+ websites, wouldn't it be a PITA if I suddenly had to
move?

**Comment by rafalkrupinski** [comment]
You're right about it being privacy and not security issue. Bad wording on my side.

Relay is a privacy tool and you're asking for a privacy anti-feature.

Vendor lock-in is inherent to Internet services and email in particular, not Relay. If I was to address the issue of Mozilla potentially abandoning Relay, I'd prefer that the foundation would pledge to transfer the domain and accounts to another provider, had they decided to discontinue the service.

**Comment by yo1dog** [comment]
You don't have to run your own email server to avoid email vendor lock-in, you just have to use a custom domain. me@gmail.com is locked to gmail because gmail owns the domain. But you can easily switch me@mydomain.com from gmail to outlook by simply updating a few DNS records for mydomain.com. This is the reason for this request: mozmail.com is vendor locked, a custom domain is not.

FWIW, I use email aliasing for spam management more so than privacy. I used to use + aliasing but that was broken as it gained popularity in the last few years. Spammers have started just sending 2 emails to both the address with and without the + alias.

Yes, it is certainly possible to detect that a certain domain is used for an extremely small subset of email addresses and to assume that all emails with that domain are the same person. But practically, I doubt that the extremely tiny percent of users that use custom domains would warrant the level of effort to actually implement that matching. It should still buy us another 10 years, just as + aliasing did. Regardless, with the rising prevalence of anonymized email (Apple "Hide My Email"), I'd guess that sophisticated fingerprint heuristics would overshadow email linking way before then anyway, if it hasn't already.
